src/lzw.ml

```diff
@@ -11,58 +20,75 @@ let rec get_bits bytedata offset count =
       let bit = (byte lsr (7 - bitoffset)) land 1 in
       (bit lsl (n - 1)) + get_bits bytedata (offset + 1) (n - 1)
 
-let build_table_entry a b =
-  match b with [] -> a | x :: _ -> List.rev (x :: List.rev a)
+let build_table_entry ~previous next =
+  if String.(equal next empty) then previous
+  else previous ^ String.make 1 next.[0]
 
 let update_buffer buffer offset codestr =
-  List.fold_left
-    (fun acc c ->
-      Cstruct.set_char buffer acc c;
-      acc + 1)
-    offset codestr
+  let len = String.length codestr in
+  Unsafe.blit_string codestr 0 buffer offset len;
+  offset + len
+
+let max_dict_size = 4096
+
+let init_table () =
+  Array.init max_dict_size (function
+    | i when i < 256 -> String.make 1 (Char.chr i)
+    | _ -> String.empty)
+
+let reset_table dict =
+  Array.iteri
+    (fun i _ ->
+      Array.set dict i
+        (if i < 256 then String.make 1 (Char.chr i) else String.empty))
```

Comment:
You don't need to reset the values under 256 - those are never be overridden.

Reply:
Fixed :)